service/lib/dinstaller/dbus/storage/proposal.rb

```diff
@@ -46,47 +53,64 @@ def initialize(logger, actions)
           dbus_reader :candidate_devices, "as"
 
           dbus_reader :available_devices, "as"
+
           # result: 0 success; 1 error
           dbus_method :Calculate, "in settings:a{sv}, out result:u" do |settings|
             backend.calculate(to_proposal_properties(settings))
 
             PropertiesChanged(INTERFACE, settings, [])
-            @actions.refresh
+            actions.refresh
 
             backend.success? ? 0 : 1
```

Comment:
Not related to this PR, but I wonder whether it might be better that `backend.calculate` already returns the success value.

Reply:
Ummm, I am not sure. I prefered to decide in the D-Bus class what value to return in case of success or failure.

Reply:
That's fine. What does not look that fine is to ask for calculation and get the success/failure value in a separate method.